espnet2/spk/pooling/chn_attn_stat_pooling.py

```diff
@@ -0,0 +1,56 @@
+import torch
+import torch.nn as nn
+
+from espnet2.spk.pooling.abs_pooling import AbsPooling
+
+
+class ChnAttnStatPooling(AbsPooling):
+    """
+    Aggregates frame-level features to single utterance-level feature.
+    Proposed in B.Desplanques et al., "ECAPA-TDNN: Emphasized Channel
+    Attention, Propagation and Aggregation in TDNN Based Speaker Verification"
+
+    args:
+        input_size: dimensionality of the input frame-level embeddings.
+            Determined by encoder hyperparameter.
+        output_size: output dimensionality of the pooled utterance-level
+            embedding. In case of this class, output_size is double of input
+            _size becuase it concatenates mean and std.
+    """
+
+    def __init__(self, input_size: int = 1536, output_size: int = 3072):
+        super().__init__()
+        self.attention = nn.Sequential(
+            nn.Conv1d(input_size * 3, 128, kernel_size=1),
+            nn.ReLU(),
+            nn.BatchNorm1d(128),
+            nn.Conv1d(128, input_size, kernel_size=1),
+            nn.Softmax(dim=2),
+        )
+
+    def output_size(self) -> int:
+        return self._output_size
```

Comment:
Now I feel this argument can be removed. This attribute has never been accessed by other modules, and it does not have any effect on the module behavior.